### PR TITLE
Use structured JSON output for local model classification

### PR DESCRIPTION
**Pull request by jdcodes1**
## Summary

- Use WebLLM's `response_format: { type: 'json_object' }` to force local models to output valid JSON instead of freeform text
- Update `parseLocalModelResponse()` to try JSON parsing first, falling back to the existing regex-based parsing for backward compatibility
- Simplify `LOCAL_SYSTEM_PROMPT` from 21 lines to 4 — format constraints are now enforced by the engine, not by prompt instructions
- Pass `responseFormat` as an optional parameter to `generate()` so non-classification callers (e.g. suggestAnnoyingReasons) still get freeform text

**Problem:** Local model responses are parsed with regex looking for "Matches X" or "No match" in freeform text. When the model phrases things differently, the regex misses and the post silently gets classified as "no match" — a false negative.

**Fix:** Constrained decoding guarantees valid JSON output. The model can only produce `{"reasoning": "...", "match": "..."}`. Zero parsing ambiguity. The vendored WebLLM (0.2.82-custom) already supports `json_object` response format — it just wasn't being used.

## Changes

| File | What |
|------|------|
| `src/background/local-model.ts` | Add `CLASSIFICATION_RESPONSE_FORMAT` constant, make `responseFormat` optional on `generate()`, pass it from `callLocalInference`, guard against `"null"` string match |
| `src/shared/prompts.ts` | Simplify `LOCAL_SYSTEM_PROMPT` (21 lines → 4 lines) |
| `tests/background/local-model.test.ts` | 6 new tests: JSON match, no-match, empty string, "null" string, regex fallback, malformed JSON fallback |

## Test plan

- [x] 6 new tests for JSON parsing + edge cases
- [x] All 205 tests pass (existing regex-based tests still pass via fallback path)
- [x] `response_format` only applied to classification calls, not suggestAnnoyingReasons
- [x] Backward compatible — non-JSON model output still parsed via regex fallback

### CODE REVIEW

File: Bouncer/src/background/local-model.ts [hunk 1]
```diff
@@ -63,6 +76,22 @@ export function parseLocalModelResponse(rawResponse: string | null): { shouldHid
...
+      const reasoning = shouldHide
+        ? `${parsed.reasoning} (Matched: ${match})`
+        : String(parsed.reasoning);
+      return { shouldHide, reasoning };
+    }
+  } catch {
+    // Not JSON — fall through to regex parsing
+  }
+
+  // Fallback: freeform text parsing (backward compatibility)
```
Comment (imperion402) [inline Bouncer/src/background/local-model.ts:94]: what is this backward compatibility you speak of?

File: Bouncer/src/shared/prompts.ts [hunk 2]
```diff
@@ -3,25 +3,9 @@
...
-<example>
-<filter_categories>politics</filter_categories>
-<post>I love cooking dinner each night with my husband</post>
-Post about making dinner at home, which is Food/lifestyle content, not politics content. No match.
-</example>
-
-You will be provided with a post (<post>) and a list of filter categories (<filter_categories>).
-Assess whether the topic of the post relates to any of the topics in the filter categories list.
-Your reasoning must be AT MOST 15 words, and MUST end with a statement of "Matches <topic>" or "No match".
+Respond with JSON: {"reasoning": "<10-15 words about what the post is about>", "match": "<matched category or null>"}
```
Comment (imperion402) [inline Bouncer/src/shared/prompts.ts:8]: what evals did you perform to validate this achieves similar metrics? What's the F1 score, accuracy, precision, etc...?

Comment (gnguralnick) [reply Bouncer/src/shared/prompts.ts:8]: this is the key point - we experimented previously with structured output like this and found that this simple of a prompt actually leads to far worse classification performance, which is why local models now receive approximately the same prompt as API ones. I'd love to be able to use a much shorter prompt if it would actually work as well since it would certainly process much faster.

**Comment by millanatimbue** [comment]
We've moved away from WebLLM and adopted a different form of structured output. Thank you for your early help!
